plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/expression/GenericRewrite.java

```diff
@@ -76,6 +87,13 @@ else if (value instanceof ConnectorExpression) {
                     if (rewritten.isEmpty()) {
                         return Optional.empty();
                     }
+                    if (checkCase && value instanceof Variable variable) {
```

Comment:
Can we move this logic to `RewriteVariable` ? And restrict them to varchar columns ?

Reply:
Or can we have a specific `Rule` for Varchar column which could capture the Variable restrict pushing down any operations on them.